### ISSUE TITLE
Adding a custom step causes a java.lang.IllegalArgumentException

### ISSUE THREAD

**Issue by honghuac** [syndesisio/syndesis#2122]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

This error appears in the integration pod log --
java.lang.IllegalArgumentException: PropertiesComponent with name properties must be defined in CamelContext to support property placeholders.

Full trace: https://gist.github.com/honghuac/ee422a0f9353be4348c7897a6efd45a2

## Expected behavior

This error should not appear, especially when FTP connector was successfully validated.

## Request and Response Data

Request -- data from http://product-catalog-lab3-product-catalog-service.apps.na1.openshift.opentlc.com/
Response -- data found on a remote host

## API Endpoints and Schemas

1. Start Connection -- HTTP Connector
2. Finish Connection -- FTP Connector 

## Tasks involved / Steps to Reproduce

1. Existing integration that works runs smoothly.
2. Then, a custom step is added between the two connectors.
3. The integration is republished.
4. Error appears in the integration pod log.
5. Publishing is unsuccessful.

**Comment by nicolaferraro** [comment]
@honghuac which custom step have you added between the connectors?

**Comment by dsimansk** [comment]
@nicolaferraro imho it's caused by changed order of steps. From the exception runtime is looking for `ftp-3.password` property. But when you create HTTP->FTP first time(Publish) and then edit property id is not increased and stays at `ftp-2.password`.

**Comment by lburgazzoli** [comment]
@dsimansk so looks like the secrets are not updated ?

**Comment by nicolaferraro** [comment]
I've just tried to reproduce but it seems to work. I get the right offsets and the secret is updated accordingly upon deployment.

I've tried with http to ftp, adding a intermediate step after first deployment. I've tried a standard log step and a extension "log body" step. Both redeployments worked correctly.

The trace clearly seems to indicate that secrets are not redeployed but it is not the case (I'm using latest build from master).
